lib/pdf_attachment_reporter.rb

```diff
@@ -0,0 +1,179 @@
+  #
+  #   - Finding the organisation which owns a PDF can be complex. In
+  #   cases where there are multiple owners, we first try to find a parent
+  #   organisation, if we know that all associated organisations are related.
+  #   Otherwise we assume that the first organisation in the list is a best
+  #   guess.
+
+  def pdfs_by_organisation
+    second_time_period_date = @last_time_period_days.days.ago.to_date
+
+    live_organisation_names = Organisation.where(govuk_status: 'live').map(&:name) << POLICY_GROUPS
+
+    live_organisation_published_pdfs_total_counts_hash = Hash[live_organisation_names.map { |o| [o, 0] }]
+    live_organisation_published_pdfs_since_first_period_counts_hash = Hash[live_organisation_names.map { |o| [o, 0] }]
+    live_organisation_published_pdfs_since_second_period_counts_hash = Hash[live_organisation_names.map { |o| [o, 0] }]
+
+    unique_published_pdf_attachments.each do |attachment|
+      pdf_attachment_data = find_pdf_attachment_data(attachment)
+
+      if pdf_attachment_data
+        live_organisation_published_pdfs_total_counts_hash[pdf_attachment_data.owning_organisation_name] += 1
+
+        if pdf_attachment_data.created_at >= second_time_period_date
+          live_organisation_published_pdfs_since_second_period_counts_hash[pdf_attachment_data.owning_organisation_name] += 1
+        end
+
+        if pdf_attachment_data.created_at >= @first_period_start_date
+          live_organisation_published_pdfs_since_first_period_counts_hash[pdf_attachment_data.owning_organisation_name] += 1
+        end
+      end
+    end
+
+    CSV.open(csv_file_path, 'wb') do |csv|
+      csv << [
+        "Organisation",
+        "Total published PDF attachments",
+        "#{@first_period_start_date} - present PDF attachments",
+        "Last #{@last_time_period_days} days PDF attachments"
+      ]
+
+      live_organisation_names.each do |organisation_name|
+        csv << [
+          organisation_name,
+          live_organisation_published_pdfs_total_counts_hash[organisation_name],
+          live_organisation_published_pdfs_since_first_period_counts_hash[organisation_name],
+          live_organisation_published_pdfs_since_second_period_counts_hash[organisation_name]
+        ]
+      end
+    end
+  end
+
+private
+
+  class PDFAttachmentData
+    attr_reader :owning_organisation_name, :created_at
+
+    def initialize(owning_organisation_name, created_at)
+      @owning_organisation_name = owning_organisation_name
+      @created_at = created_at
+    end
+  end
+
+  def find_pdf_attachment_data(attachment)
+    attachable_object = find_attachable_object(attachment)
+
+    if attachable_object
+      if attachable_object.class == PolicyGroup
+        PDFAttachmentData.new(POLICY_GROUPS, attachment.created_at)
+      else
+        # Responses are only sometimes linked to organisations (via a consultation)
+        if attachable_object.is_a?(Response)
```

Comment:
...or this one a `.class == Response` (I suspect the `.is_a?(...)` version is a better choice as there might be subclasses of `Response` that you need to deal with.